task20/js/main.js

```diff
@@ -0,0 +1,88 @@
+import '../css/style.css'
+import { createElementWithClassName } from './utils/createElement'
+import { renderLoading } from './modules/loading'
+import { removeLoading } from './modules/loading'
+import { renderErrorMessage } from './modules/error'
+
+const endpoint = "https://mocki.io/v1/7afded4f-46ed-4535-b770-997abc352aea";
+
+const parent = document.getElementById("js-parent");
+
+const getFetchData = async () => {
+    const response = await fetch(endpoint);
+    if (!response.ok){
+        console.error(`$(response:status):$(response:statusText)`);
+        renderErrorMessage(parent, "問題が発生し表示することができません。");
+    }
+    const json = await response.json();
+    return json.data;
+};
+
+const fetchTableData = () => {
+    return new Promise((resolve) => {
+        setTimeout(() => {
+            resolve(getFetchData(endpoint));
+        }, 3000);
+    });
+};
+
+const initialize = async () => {
+    renderLoading(parent);
+    let userContentsData
+    try {
+        userContentsData = await fetchTableData();
+    } catch(error) {
+        renderErrorMessage(parent, "問題が発生し表示することができません。");
+    } finally {
+        removeLoading(parent);
+    }
+    renderTableContents(userContents, userContentsData);
+};
+initialize();
+
+const userContents =  {
+    "userId" : "ID",
+    "userName" : "名前",
+    "userGender" : "性別",
+    "userAge" : "年齢"
+};
```

Comment:
IMO
For tables, I was taught that it is common to use ``column``. ``userTableColumn`` or ``userDataColumn`` may be appropriate names.

Reply:
I learned a lot because the word "column" was not in my field of vision. Thank you very much.
[6280314](https://github.com/seeeen044/JavaScript_handson/pull/21/commits/62803142abc3f91a585e303be3c0ce5800e10e33)